### PR TITLE
Finish recipe repository class

### PR DESCRIPTION
**Pull request by cagallo**
#### What's this PR do?

- Finish filterByTags method

#### Where should the reviewer start?

- Recipe-repository.js 

#### How should this be manually tested?

- N/A

#### What are the relevant tickets?

-  Get recipe.js done, out of iteration 1!

#### Screenshots (if appropriate)

- N/A

#### Questions:

- N/A

### CODE REVIEW

File: test/RecipeRepository-test.js [hunk 1]
```diff
@@ -355,10 +355,9 @@ describe('RecipeRepository', () => {
     expect(repository.recipeData).to.equal(sampleData);
   });
 
-  it.skip('should return a list of recipes based on one or more tags', () => {
-    const recipesByTag = repository.filterByTag('sauce');
-
-    expect(recipesByTag).to.equal(recipeData[2]);
+  it('should return a list of recipes based on one or more tags', () => {
+    const recipesByTag = repository.filterByTags(['sauce']);
```
Comment (anewb87) [inline test/RecipeRepository-test.js:359]: Bingo! This is an array, not a string.

File: src/classes/RecipeRepository.js [hunk 2]
```diff
@@ -59,18 +59,27 @@ class RecipeRepository {
 
     return ingredientID;
   };
+
+  filterByTags(selectedTags) {
+    return this.recipeData.reduce((acc, recipe) => {
+      recipe.tags.forEach(tag => {
+        selectedTags.forEach(selectedTag => {
```
Comment (anewb87) [inline src/classes/RecipeRepository.js:66]: Nested forEach().
Super slick.

**Review by anewb87 — APPROVED** [review]
Yes! Our first reduce!

**Review by kammon10 — APPROVED** [review]
looks fantastic!

**Review by aliroemhildt — APPROVED** [review]

**Pull request merged**